### PR TITLE
Volmgr fixes

### PR DESCRIPTION
**Pull request by zedi-pramodh**
This is same as PR https://github.com/naiming-zededa/eve/pull/354 accept on poc-oct3  branch and  addressed review comments there.

Lots of code clean up.
Added IsLocal flag to content tree config and status. That flag will tell remote nodes if the content needs to be downloaded or not.
Added IsReplicated and IsNativeContainer flag to volumeconfig and volume status. IsReplicated flag will tell the remote nodes if the PVC is replicated. All replicated volumes are handled differently, ie no garbage collection, no delete PVC etc.
IsNativeContainer flag is used not to create a PVC for container image.

Added IsDesignatedNodeID to appinstanceconfig.

We parse is designated node id from volumeconfig, content tree config and apinstance config in zedagent code.

Tested:

VM and container failover and fallback multiple times. Seems to work.

### CODE REVIEW

**Review by andrewd-zededa — COMMENTED** [review]
Are there any docs to help follow the IsLocal and IsReplicated states and their lifecycles?

File: pkg/pillar/cmd/zedagent/parseconfig.go [hunk 1]
```diff
@@ -652,6 +652,8 @@ func parseAppInstanceConfig(getconfigCtx *getconfigContext,
 		appinstancePrevConfigHash, configHash, Apps)
 	appinstancePrevConfigHash = configHash
 
+	devUUIDStr, _ := os.Hostname()
```
[the remark this answers is not in the document]
Comment (zedi-pramodh) [reply pkg/pillar/cmd/zedagent/parseconfig.go:655]: I don't think so. Since the App config processing way after onboarding of the device.

**Comment by zedi-pramodh** [comment]
> Are there any docs to help follow the IsLocal and IsReplicated states and their lifecycles?

While sending reviews to master  I will add a doc file.

Basically IsReplicated means the volume designated node id is not my node. That helps not create volume just wait for PVC to show up.

IsLocal is for content tree. Same concept here too. If content tree is set to isLocal that node is designated node and the content will be downloaded. Only exception is native containers. In that case we download content (container image) to all nodes in cluster.

If content tree is not IsLocal nothing gets downloaded on that node and we except to have replicated PVC on that node with the content.

**Comment by zedi-pramodh** [comment]
Someone bless this one please, unless there are concerns.  I did test failovers as much as possible. We need to have eve build ready by Monday to test end to end.

**Review by naiming-zededa — APPROVED** [review]

**Pull request merged**